cypress-tests/cypress/e2e/v2/searchEntities.cy.ts

```diff
@@ -0,0 +1,180 @@
+    second: `${stringProject}-${stringSecond}-${stringRandomTwo}-${getRandomString()}`,
+  };
+  const groups = {
+    first: `${stringGroup}-${stringFirst}-${stringRandomTwo}-${getRandomString()}`,
+    second: `${stringGroup}-${stringSecond}-${stringRandomOne}-${getRandomString()}`,
+  };
+  let userNamespace: string;
+
+  // Create the requires resources
+  before(() => {
+    // Login
+    cy.session(
+      sessionId,
+      () => {
+        cy.robustLogin();
+      },
+      validateLoginV2,
+    );
+
+    // Create groups and projects -- fail early if any of the resources are not created
+    getUserData().then((user: User) => {
+      userNamespace = user.username;
+      for (const proj of [projects.first, projects.second]) {
+        createProjectIfMissingAPIV2({
+          name: proj,
+          namespace: user.username,
+          slug: proj,
+        }).then((response) => {
+          // eslint-disable-line max-nested-callbacks
+          if (response.status >= 400) {
+            throw new Error("Failed to create project");
+          }
+        });
+      }
+      for (const grp of [groups.first, groups.second]) {
+        createGroupIfMissingAPI({
+          name: grp,
+          slug: grp,
+        }).then((response) => {
+          // eslint-disable-line max-nested-callbacks
+          if (response.status >= 400) {
+            throw new Error("Failed to create group");
+          }
+        });
+      }
+    });
+  });
+
+  // Restore the session (login)
+  beforeEach(() => {
+    cy.session(
+      sessionId,
+      () => {
+        cy.robustLogin();
+      },
+      validateLoginV2,
+    );
+  });
+
+  // Cleanup the resources after the test
+  after(() => {
+    [projects.first, projects.second].forEach((proj) => {
+      getProjectByNamespaceAPIV2({
+        slug: proj,
+        namespace: userNamespace,
+      }).then((response) => {
+        if (response.status === 200) {
+          deleteProjectFromAPIV2({
+            id: response.body.id,
+            slug: proj,
+          });
+        }
+      });
+    });
+    [groups.first, groups.second].forEach((grp) => {
+      getGroupFromAPI(grp).then((response) => {
+        if (response.status === 200) {
+          deleteGroupFromAPI(grp);
+        }
+      });
+    });
+  });
+
+  it("Can search for entities and filter works", () => {
+    cy.visit("/v2");
+
+    // Search for string
+    cy.getDataCy("navbar-link-search").click();
+    cy.intercept(
+      new RegExp(
+        `(?:/ui-server)?/api/search/query\\?q=.*?${stringRandomOne}.*`,
+      ),
+    ).as("searchQueryOne");
+    cy.getDataCy("search-input").clear().type(stringRandomOne);
+    cy.getDataCy("search-button").click();
+    cy.wait("@searchQueryOne");
+    cy.getDataCy("search-card").should("have.length", 2);
+    cy.intercept(
+      new RegExp(
+        `(?:/ui-server)?/api/search/query\\?q=.*?${stringRandomTwo}.*`,
+      ),
```

Comment:
question/suggestion: If you intercept a less specific path to the search api isn't it easier? I.e. if you intercept all requests to the search api then you can always just wait for those interceptions to complete and you would not have `@searchQueryOne` and `@searchQueryTwo`? You can just wait for `@searchQuery`.

Reply:
You're right, no need to differentiate. Here I used a generic matcher 893dff814f5af245d8ba87291c206f98f313f061